parsers/niederbayern_oberpfalz.py

```diff
@@ -61,6 +31,26 @@
 from pyopenmensa.feed import LazyBuilder
 
 
+# Only allow a newline after 8 separators
+def replace_invalid_newlines(s):
```

Comment:
This can be fixed much simpler with the following regex, which matches one or more newlines, if they are not followed by the date pattern and semicolon (negative lookahead): `\n+(?!\d\d\.\d\d\.\d\d\d\d;)`

For reference, the problematic part looks like:
```
12.11.2019;Di;Suppe;Grießnockerlsuppe (3,A,AA,C,I);V;0,70 / 0,90 / 1,40;0,70;0,90;1,40
12.11.2019;Di;HG1;// Selbstbedienung //

Schinkennudeln (2,3,8,A,AA);S;2,00 / 2,80 / 3,60;2,00;2,80;3,60
12.11.2019;Di;HG2;Vegane Kartoffelpizza (3,F);VG;2,80 / 3,40 / 4,40;2,80;3,40;4,40
```